kong/llm/drivers/gemini.lua

```diff
@@ -292,7 +292,24 @@ function _M.subrequest(body, conf, http_opts, return_res_table)
     ["Content-Type"] = "application/json",
   }
 
-  if conf.auth and conf.auth.header_name then
+  if identity_interface and identity_interface.interface then
+    if identity_interface.interface:needsRefresh() then
+      -- HACK: A bug in lua-resty-gcp tries to re-load the environment
+      --       variable every time, which fails in nginx
+      --       Create a whole new interface instead.
+      --       Memory leaks are mega unlikely because this should only
+      --       happen about once an hour, and the old one will be
+      --       cleaned up anyway.
+      local service_account_json = identity_interface.interface.service_account_json
+      local identity_interface_new = identity_interface.interface:new(service_account_json)
```

Comment:
will :new at run-time  being  affected by the fact that env vars are not variable during run-time?

Reply:
Yeah it does, but actually this code is to get around this limitation!

So if you just want to call the refresh token, it doesn't work because environment variable is empty. So I:

1. Rip the service account JSON from the old interface, into a new one
2. Force it to authenticate
3. Get the token back out the new interface, and into the old interface
4. Destroy the new interface

It's a hack, but it definitely works, but I found this late and I can't make changes to the GCP library and re-test everything in this release.